automod/rules/replies.go

```diff
@@ -27,3 +27,24 @@ func ReplyCountPostRule(evt *automod.RecordEvent, post *appbsky.FeedPost) error
 	evt.IncrementDistinct("reply-to", did, parentURI.Authority().String())
 	return nil
 }
+
+var identicalReplyLimit = 5
+
+// Looks for accounts posting the exact same text multiple times. Does not currently count the number of distinct accounts replied to, just counts replies at all.
+//
+// There can be legitimate situations that trigger this rule, so in most situations should be a "report" not "label" action.
+func IdenticalReplyPostRule(evt *automod.RecordEvent, post *appbsky.FeedPost) error {
+	if post.Reply == nil || IsSelfThread(evt, post) {
+		return nil
+	}
+
+	// use a specific period (IncrementPeriod()) to reduce the number of counters (one per unique post text)
+	period := automod.PeriodDay
+	bucket := evt.Account.Identity.DID.String() + "/" + HashOfString(post.Text)
+	if evt.GetCount("reply-text", bucket, period) >= identicalReplyLimit {
```

Comment:
Right now, this is proceeding to act immediately on a hash collision.  Do you think it would be reasonable to do a more expensive check to see if things are *actually* identical?

I don't know how bad a false positive is here.  Maybe if the threshhold for identical in sheer count is moderate, it's unlikely to trigger on reasonable real human behavior?

Reply:
I think the naive false positive rate (for the 64bit variant of murmur3) is low enough to not worry about it and not do secondary network requests to check for exact matches.

This isn't a cryptographic hash, so attacks could be a concern for some rules.

Generally, I feel like all the counters we are using here should be treated as a bit fuzzy, at least for record-level counts. It is totally possible for events to get partially-processed (and partially persisted) and then re-processed again after a crash. I think the semantics and kinds of rules and actions we write are generally resilient to this: doing things like reporting for human review, or having large margins before taking fully-automated action.